tap_shopify/streams/base.py

```diff
@@ -204,12 +206,16 @@ def get_objects(self):
                     objects = self.call_api(query_params)
 
                 for obj in objects:
+                    dict_obj = obj.to_dict()
+                    replication_value = strptime_to_utc(dict_obj[self.replication_key])
                     if obj.id < since_id:
                         # This verifies the api behavior expectation we
                         # have that all results actually honor the
                         # since_id parameter.
                         raise OutOfOrderIdsError("obj.id < since_id: {} < {}".format(
                             obj.id, since_id))
+                    if replication_value > max_bookmark:
+                        max_bookmark = replication_value
```

Comment:
# Opinion

 I think we should move `replication_value`'s initialization to just above line 217 here.

It's not used in lines 211 to 216 and only used in this area to check if we should updated `max_bookmark`

Reply:
Moved the initialization of `replication_value` above line 217